FreeRTOS/FreeRTOS-Plus-TCP
# Handle buffer allocation failure in `prvCreateSectors`

Description
-----------
This PR fixes an ignored memory allocation failure in `prvCreateSectors` by checking the return values and passing them to caller APIs. Thanks, @hlef for reporting this issue in #1149.

Test Steps
-----------

PR tests:
HW tests on STM32F4 DUT: with device as both TCP client and server
CI demo runs: https://github.com/tony-josi-aws/FreeRTOS/actions/runs/9383436139/job/25837038341

Checklist:
----------



Related Issue
-----------
#1149 


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
